src/rpm.rs

```diff
@@ -44,7 +52,112 @@ impl RPMPackage {
         out.write_all(&self.content)?;
         Ok(())
     }
+
+    // TODO allow passing an external signer/verifier
+
+    /// sign all headers (except for the lead) using an external key and store it as the initial header
+    #[cfg(feature = "signing-meta")]
+    pub fn sign<S>(&mut self, secret_key: &[u8]) -> Result<(), RPMError>
```

Comment:
I think we should align this method with the builder and only pass a signer directly so it can be like:

```rust
 pub fn sign<S,A>(&mut self, signer: S) -> Result<(), RPMError>
    where
        S: crypto::Signing<A>,
        A: crypto::algorithm::Algorithm,
```

This way the signing process is consistent and the api is flexible enough to handle signers with various configurations.